apps/angular/crud/src/app/components/app.component.ts

```diff
@@ -0,0 +1,50 @@
+import { CommonModule } from '@angular/common';
+import { Component, DestroyRef, OnInit, inject } from '@angular/core';
+import { TodoItemComponent } from './todo-item/todo-item.component';
+import { Todo } from '../types/todo';
+import { TodoService } from '../services/todo.service';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Observable } from 'rxjs';
+
+@Component({
+  standalone: true,
+  selector: 'app-root',
+  template: `
+    <app-todo-item
+      *ngFor="let todo of todosList$ | async"
+      [todo]="todo"
+      (updateTodo)="onTodoUpdate($event)"
+      (deleteTodo)="onTodoDelete($event)"></app-todo-item>
+  `,
+  styles: [],
+  imports: [CommonModule, TodoItemComponent],
+  providers: [TodoService],
+})
+export class AppComponent implements OnInit {
+  private readonly todoService = inject(TodoService);
+  private readonly destroyRef = inject(DestroyRef);
+
+  todosList$!: Observable<Todo[]>;
+
+  ngOnInit(): void {
+    this.todoService
+      .initTodos()
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe();
+    this.todosList$ = this.todoService.todoList$;
+  }
+
+  onTodoUpdate(todo: Todo) {
+    this.todoService
+      .updateTodo(todo)
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe();
+  }
+
+  onTodoDelete(todo: Todo) {
+    this.todoService
+      .deleteTodo(todo)
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe();
+  }
```

Comment:
it's nice to use new operator like `takeUntilDestroyed` but in your case, you are interesting in only the first emitted value so use `take(1)` because in your case, your observable will never get destroy (not really true because you are calling a http request with complete on its own but you need to be careful) and everytime you click on a button, a new observable will occur.

Reply:
Thank you for the insightful comment. I didn't consider that the destruction of the component may never occur. 
Also, as i see, you didn't comment on other people not unsubscribing from http requests. Isn't it causing memory leaks or in current angular versions there is no need for unsubscriptions from them?

Reply:
That's a very good remark. 
Indeed I don't comment other person PR because it's better practise to unsuscribe but it will not leak because http will complete when the request respond. 
The only thing that can happen is if you quit a component and the request take time, you can have weird side effect because the observable is not unsuscribe.

Reply:
Thank you very much for your answers